tests/OAuthTwoTest.php

```diff
@@ -24,17 +25,67 @@ protected function tearDown(): void
         m::close();
     }
 
-    public function testRedirectGeneratesTheProperIlluminateRedirectResponse()
+    public function testRedirectGeneratesTheProperIlluminateRedirectResponseWithoutPKCE()
     {
         $request = Request::create('foo');
         $request->setLaravelSession($session = m::mock(Session::class));
-        $session->shouldReceive('put')->once();
+
+        $state = null;
+        $closure = function($name, $stateInput) use(&$state) {
+            if($name == 'state') {
+                $state = $stateInput;
+                return true;
+            } else {
+                return false;
+            }
+        };
+
+        $session->shouldReceive('put')->once()->withArgs($closure);
         $provider = new OAuthTwoTestProviderStub($request, 'client_id', 'client_secret', 'redirect');
         $response = $provider->redirect();
 
         $this->assertInstanceOf(SymfonyRedirectResponse::class, $response);
         $this->assertInstanceOf(RedirectResponse::class, $response);
-        $this->assertSame('http://auth.url', $response->getTargetUrl());
+        $this->assertSame('http://auth.url?client_id=client_id&redirect_uri=redirect&scope=&response_type=code&state='.$state, $response->getTargetUrl());
+    }
+
+    private static $codeVerifier = null;
```

Comment:
I was unclear on how to get this value stored in the test in a local variable, it seemed to only work when stored as a static class variable. The test needs to know the value the main library generated, and also needs to be able to return it to the main library when pulled from the session. Let me know if there is a better way to do this, it was very difficult to find any docs on how to do this kind of test.

Reply:
You should be able to move this into the test method itself like this:

```php
public function testRedirectGeneratesTheProperIlluminateRedirectResponseWithPKCE()
{
    static $codeVerifier = null;
```

Reply:
Strangely that isn't working. The first closure sets that value, but then it appears to be null by the time the second closure goes to read it.